enterprise/frontend/src/metabase-enterprise/google_drive/GdriveConnectionModal.tsx

```diff
@@ -41,10 +41,17 @@ export function GdriveConnectionModal({
   onClose: () => void;
   reconnect: boolean;
 }) {
+  const shouldShow = useShowGdrive();
   const { data: { email: serviceAccountEmail } = {} } =
-    useGetServiceAccountQuery();
+    useGetServiceAccountQuery(shouldShow ? undefined : skipToken);
```

Comment:
nit: This feels notation feels awkward. There is another option, but I don't know that it's any better haha
```
useGetSetviceAccountQuery(undefined, {skip: !shouldShow})
```